vendor/github.com/Azure/go-autorest/autorest/sender.go

```diff
@@ -215,20 +215,26 @@ func DoRetryForStatusCodes(attempts int, backoff time.Duration, codes ...int) Se
 			rr := NewRetriableRequest(r)
 			// Increment to add the first call (attempts denotes number of retries)
 			attempts++
-			for attempt := 0; attempt < attempts; attempt++ {
+			for attempt := 0; attempt < attempts; {
 				err = rr.Prepare()
 				if err != nil {
 					return resp, err
 				}
 				resp, err = s.Do(rr.Request())
-				// we want to retry if err is not nil (e.g. transient network failure)
-				if err == nil && !ResponseHasStatusCode(resp, codes...) {
+				// we want to retry if err is not nil (e.g. transient network failure).  note that for failed authentication
+				// resp and err will both have a value, so in this case we don't want to retry as it will never succeed.
+				if err == nil && !ResponseHasStatusCode(resp, codes...) || IsTokenRefreshError(err) {
 					return resp, err
 				}
 				delayed := DelayWithRetryAfter(resp, r.Cancel)
 				if !delayed {
 					DelayForBackoff(backoff, attempt, r.Cancel)
 				}
+				// don't count a 429 against the number of attempts
+				// so that we continue to retry until it succeeds
+				if resp.StatusCode != http.StatusTooManyRequests {
```

Comment:
👋🏻 @jhendrixMSFT @marstr we're seeing a crash here - we handle checking for this internally to see if there was a dropped connection from Azure (which happens often enough that we need to handle it) - any chance that we could check if `resp != nil` before checking the status code?

Reply:
I will get this fixed and release a v9.5.2 today.

Reply:
awesome - thanks @jhendrixMSFT :)

Reply:
The fix is in an the release has been tagged.  https://github.com/Azure/go-autorest/releases/tag/v9.5.2